sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFileFormat.scala

```diff
@@ -207,13 +210,23 @@ class ParquetFileFormat
 
       val sharedConf = broadcastedHadoopConf.value.value
 
-      val fileFooter = if (enableVectorizedReader) {
+      val (inputFile: Option[HadoopInputFile],
+          inputStream: Option[SeekableInputStream],
+          fileFooter: ParquetMetadata) = if (enableVectorizedReader) {
         // When there are vectorized reads, we can avoid reading the footer twice by reading
         // all row groups in advance and filter row groups according to filters that require
         // push down (no need to read the footer metadata again).
-        ParquetFooterReader.readFooter(sharedConf, file, ParquetFooterReader.WITH_ROW_GROUPS)
+        val hadoopInputFile = HadoopInputFile.fromStatus(file.fileStatus, sharedConf)
+        val fileInputStream = hadoopInputFile.newStream()
+        val footerFilter = ParquetFooterReader.footerFilter(
+          sharedConf, file, ParquetFooterReader.WITH_ROW_GROUPS)
+        val footer = ParquetFooterReader.readFooter(
+          hadoopInputFile, fileInputStream, footerFilter)
```

Comment:
TODO: close the reader but keep the `fileInputStream` open, waiting for https://github.com/apache/parquet-java/pull/3208